tests/ops/op_math/test_sprod.py

```diff
@@ -0,0 +1,635 @@
+        assert sprod_op.parameters == [[3.14], [0.23]]
+        assert sprod_op.num_params == 2
+
+    def test_queue_idx(self):
+        sprod_op = s_prod(3.14, qml.Identity(wires=1))
+        assert sprod_op.queue_idx is None
+
+    def test_parameters(self):
+        sprod_op = s_prod(9.87, qml.Rot(1.23, 4.0, 5.67, wires=1))
+        assert sprod_op.parameters == [[9.87], [1.23, 4.0, 5.67]]
+
+    def test_data(self):
+        sprod_op = s_prod(9.87, qml.Rot(1.23, 4.0, 5.67, wires=1))
+        assert sprod_op.data == [[9.87], [1.23, 4.0, 5.67]]
+
+    def test_data_setter(self):
+        """Test the setter method for"""
+        scalar, angles = (9.87, (1.23, 4.0, 5.67))
+        old_data = [[9.87], [1.23, 4.0, 5.67]]
+
+        sprod_op = s_prod(scalar, qml.Rot(*angles, wires=1))
+        assert sprod_op.data == old_data
+
+        new_data = [[1.23], [0.0, -1.0, -2.0]]
+        sprod_op.data = new_data
+        assert sprod_op.data == new_data
+        assert sprod_op.scalar == new_data[0][0]
+        assert sprod_op.base.data == new_data[1]
+
+    @pytest.mark.parametrize("scalar, op", ops)
+    def test_terms(self, op, scalar):
+        sprod_op = SProd(scalar, op)
+        coeff, ops = sprod_op.terms()
+
+        assert coeff == [scalar]
+        for op1, op2 in zip(ops, [op]):
+            assert op1.name == op2.name
+            assert op1.wires == op2.wires
+            assert op1.data == op2.data
+
+    def test_decomposition_raises_error(self):
+        sprod_op = s_prod(3.14, qml.Identity(wires=1))
+
+        with pytest.raises(DecompositionUndefinedError):
+            sprod_op.decomposition()
+
+
+class TestMscMethods:
+    """Test miscellaneous methods of the SProd class."""
+
+    @pytest.mark.parametrize("op_scalar_tup, op_rep", tuple((i, j) for i, j in zip(ops, ops_rep)))
+    def test_repr(self, op_scalar_tup, op_rep):
+        """Test the repr dunder method."""
+        scalar, op = op_scalar_tup
+        sprod_op = SProd(scalar, op)
+        assert op_rep == sprod_op.__repr__()
+
+    @pytest.mark.parametrize("op_scalar_tup", ops)
+    def test_copy(self, op_scalar_tup):
+        """Test the copy dunder method properly copies the operator."""
+        scalar, op = op_scalar_tup
+        sprod_op = SProd(scalar, op)
+        copied_op = copy(sprod_op)
+
+        assert sprod_op.scalar == copied_op.scalar
+
+        assert sprod_op.id == copied_op.id
+        assert sprod_op.data == copied_op.data
+        assert sprod_op.wires == copied_op.wires
+
+        assert sprod_op.base.name == copied_op.base.name
+        assert sprod_op.base.wires == copied_op.base.wires
+        assert sprod_op.base.data == copied_op.base.data
+
+
+class TestMatrix:
+    @pytest.mark.parametrize("scalar", scalars)
+    @pytest.mark.parametrize("op_and_mat", non_param_ops)
+    def test_non_parametric_ops(self, scalar, op_and_mat):
+        """Test matrix method for a scalar product of non_parametric ops"""
+        op, mat = op_and_mat
```

Comment:
```suggestion
    @pytest.mark.parametrize("op, mat", non_param_ops)
    def test_non_parametric_ops(self, scalar, op, mat):
        """Test matrix method for a scalar product of non_parametric ops"""
```

You can repeat that through other tests in this section.

Reply:
got it 👍🏼